### DISCUSSION TITLE
Is there an API call to get server metadata?

### DISCUSSION THREAD

**Discussion by GreenFootballs** [bluesky-social/atproto#1139] [How to do it?] [answered]
Is there an API endpoint that returns info about server constraints like maximum image size, text length, etc.?

I've looked through the atproto API docs and if it's there I must have missed it.

**Comment by Matrix89** [comment]
AFAIK nope, there's a health endpoint but there's no endpoint for getting the metadata, although it sounds useful

**Comment by devinivy** [comment]
There is an endpoint `com.atproto.server.describeServer` that provides details about the server.  It doesn't contain information like max upload size, but if the server were to publicize that info I believe this is where it would go.

Also worth noting that the lexicons for an application might define max text sizes and max image sizes on each field of a record.  You can see a good example of that here for user profiles: https://github.com/bluesky-social/atproto/blob/bfd0670d38b7ab59f119ee9907b0cea7bf5f2374/lexicons/app/bsky/actor/profile.json#L16-L25

**Comment by GreenFootballs** [comment]
Yes, I've seen those maxSize properties in the lexicons, but I was hoping there might be an endpoint a client could use on an automatic basis.

I know there's only one server now, but couldn't these values change on other servers?

**Comment by devinivy** [comment, marked as answer]
The `app.bsky.*.*` lexicons are independent of any PDS, instead tied to an application (Bluesky Social).  In order to maintain high compatibility throughout the ecosystem, application lexicons should not vary by server (barring that different participants in the network may have more or less up-to-date versions of a given schema).  For example, the purpose of the app.bsky lexicon is to establish lexicons for microblogging.  If some servers decide that posts can be  10000 characters long and others decide that posts can only be 300 characters long, then the application begins to have an ambiguous purpose and clients would have trouble working for folks on many different servers.  I could still imagine some servers having their own max upload size limit or max repository sizes, independent from any application.

**Reply by kjarex** [reply]
Not sure which one is right, but the answer here is contradicting the answer I received in https://github.com/bluesky-social/atproto/discussions/1582

I would actually also understand it the way as it's said here (for the same reasoning about what should be pds dependent/independent). 

When I found this discussion, I thought my question (=the other discussion) has been already discussed and answered here, but as the answers are contradicting, it might be still unclear and in need of a final decission

**Reply by snarfed** [reply]
@kjarex hmm! What's the contradiction?
